src/js/system/RoutingService.js

```diff
@@ -0,0 +1,113 @@
+import {Route} from 'react-router';
+
+const queue = {};
+
+class Tab {
+  constructor(path, component) {
+    this.path = path;
+    this.component = component;
+  }
+
+  getPath() {
+    return this.path;
+  }
+
+  getComponent() {
+    return this.component;
+  }
+}
+
+class Page {
+  constructor(path, component) {
+    this.path = path;
+    this.component = component;
+    this.tabs = [];
+  }
+
+  getPath() {
+    return this.path;
+  }
+
+  getComponent() {
+    return this.component;
+  }
+
+  getTabs() {
+    return this.tabs;
+  }
+
+  addTab(path, component) {
+    const tab = new Tab(path, component);
+    this.tabs.push(tab);
+  }
+}
+
+const RoutingService = {
```

Comment:
*Side Note: We need to think about a good singleton pattern, as we use various different ways of defining singletons and they all have some flaws.*